Source/engine/render/text_render.cpp

```diff
@@ -631,8 +631,11 @@ uint32_t DrawString(const Surface &out, string_view text, const Rectangle &rect,
 	else if (HasAnyOf(flags, UiFlags::AlignRight))
 		characterPosition.x += rect.size.width - lineWidth;
 
+	auto textLines = (int)std::count(text.begin(), text.end(), '\n');
+	int yMax = rect.position.y + rect.size.height * (textLines + 1);
+
 	int rightMargin = rect.position.x + rect.size.width;
-	const int bottomMargin = rect.size.height != 0 ? std::min(rect.position.y + rect.size.height, out.h()) : out.h();
+	const int bottomMargin = rect.size.height != 0 ? std::min(yMax, out.h()) : out.h();

```

Comment:
The rectangle defines the bounds for the string. It shouldn't depend on the number of newlines in the input text. Rather, it's up to the caller to determine what the appropriate bounds should be for the string that will be drawn.

I believe what you're going for here should be done by defining the full rectangle for the region inside the info panel and then using `UiFlags::VerticalCenter`.

Reply:
Worked better without the additional UI flag